File: core/src/main/java/com/serious/service/ComponentInterface.java
package com.serious.service;
/*
 * @COPYRIGHT (C) 2023 Andreas Ernst
 *
 * All rights reserved
 */

import java.lang.annotation.*;

/**
 * @author Andreas Ernst
 */
@Documented
@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.TYPE)
public @interface ComponentInterface {
    String name() default "";
    String description() default "";

    Class<? extends Service>[] services();
}

File: core/src/main/java/com/serious/service/ComponentInterface.kt
package com.serious.service
/*
 * @COPYRIGHT (C) 2023 Andreas Ernst
 *
 * All rights reserved
 */

import kotlin.reflect.KClass


/**
 * Marks [Component] interfaces
 */
@MustBeDocumented
@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.CLASS)
annotation class ComponentInterface(
    val name: String = "",
    val description: String = "",
    val services: Array<KClass<out Service>>
) 